rfc/graphql-schema/additionalTypeDefs/byPage/invoices.ts

```diff
@@ -0,0 +1,143 @@
+export default /* GraphQL */ `
+  enum InvoiceStatus {
+    OPEN
+    PAID
+    OVERDUE
+    CANCELLED
+  }
+
+  type InvoiceStats {
+    openBalance: Float!
+    overdueBalance: Float!
+  }
+
+  input InvoiceDateRangeFilterInput {
+    from: DateTime!
+    to: DateTime
+  }
+
+  input InvoiceFiltersInput {
+    search: String
+    dateRange: InvoiceDateRangeFilterInput
+    status: InvoiceStatus
+  }
+
+  enum InvoicesSortInput {
+    INVOICE_NUMBER_A_TO_Z
+    INVOICE_NUMBER_Z_TO_A
+    ORDER_ID_A_TO_Z
+    ORDER_ID_Z_TO_A
+    CREATED_AT_NEWEST
+    CREATED_AT_OLDEST
+    DUE_DATE_NEWEST
+    DUE_DATE_OLDEST
+    ORIGINAL_BALANCE_HIGHEST
+    ORIGINAL_BALANCE_LOWEST
+    OPEN_BALANCE_HIGHEST
+    OPEN_BALANCE_LOWEST
+  }
+
+  type CreateCartFromInvoiceError implements Error {
+    message: String!
+  }
+
+  type CartCreateResult {
+    cart: Cart
+    errors: [CreateCartFromInvoiceError!]!
+  }
+
+  type GenerateInvoicePdfError implements Error {
+    message: String!
+  }
+
+  type GenerateInvoicePdfResult {
+    url: String
+    errors: [GenerateInvoicePdfError!]!
+  }
+
+  type Invoice {
+    id: ID!
+    invoiceNumber: String!
+    createdAt: DateTime!
+    dueDate: DateTime!
+    openBalance: Money!
+    originalBalance: Money!
+    status: InvoiceStatus!
+    order: Order!
+    company: Company!
+  }
+
+  type InvoiceEdge {
+    node: Invoice!
+    cursor: String!
+  }
+
+  type InvoiceConnection {
+    edges: [InvoiceEdge!]!
+    pageInfo: PageInfo!
+    collectionInfo: CollectionInfo!
+    stats: InvoiceStats!
```

Comment:
`stats` will be an odd one out amongst all our `XConnection` types

what is it detailing, the total costs of all outstanding invoices?


we could avoid this with something like

```gql
type InvoiceTotals {
  openBalance: Float!
  overdueBalance: Float!
}

extend type Company {
  # etc etc
  invoiceTotals: InvoiceTotals!
}
```